NitroxPatcher/Patches/Stalker_CheckLoseTooth_Patch.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Harmony;
+using NitroxModel.Helper;
+using System;
+using System.Reflection;
+using UnityEngine;
+
+namespace NitroxPatcher.Patches
+{
+    public class Stalker_CheckLoseTooth_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(Stalker);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("CheckLoseTooth", BindingFlags.NonPublic | BindingFlags.Instance);
+
+        public static bool Prefix(Stalker __instance, GameObject target)
+        {
+            float hardness = 0f;
+            TechType techType = CraftData.GetTechType(target);
+            LiveMixin liveMixin = target.GetComponent<LiveMixin>();
+            
+            if (techType == TechType.ScrapMetal || techType == TechType.Titanium
+             || techType == TechType.Seamoth || techType == TechType.Exosuit)
+            {
+                hardness = 0.3f; //15% probability
+            }
+            else if(liveMixin != null && liveMixin.IsAlive())
+            {
+                hardness = 0.1f; //5% probability
+            }
+
+            if (UnityEngine.Random.value < hardness && UnityEngine.Random.value < 0.5f)
```

Comment:
Why do we have `UnityEngine.Random.value < 0.5f`?  For the tooth to drop it would have to be less than the hardnesses above (0.3 & 0.1) So it seems that doesn't do anything.

Reply:
It does, although it could be expressed in one random, those are two different Randoms. It means that it always drops half the time it would drop. So, statistically it's the same as setting 15% and 5%, but stating clearly that it will never drop for sure. E.g. hardness of 1f will drop 50% of the time.

Reply:
Ahh, I get it.  Let's express in one random.